frontend/components/LoadGraph.jsx

```diff
@@ -1,64 +1,124 @@
-import React from 'react';
-import { Chart } from 'react-charts';
+import { Box } from '@material-ui/core';
+import React, { useEffect, useState } from 'react';
+import Chart from 'react-apexcharts';
 
-import useDemoConfig from './GraphUtils/useDevConfig';
-import ResizableBox from './GraphUtils/ResizableBox';
-import { Command } from './CommandWidgets';
-import { Box, Typography } from '@material-ui/core';
+const command = 'uptime';
+const interval = 1000;
 
 export default function LoadGraph() {
-  const datasets = {
-    labels: [],
-    data: [{}],
-  };
-
-  const { data, updateData } = useDemoConfig({
-    series: 3,
+  const [state, setstate] = useState({
+    data: [[], [], []],
+    output: '',
   });
 
-  const series = React.useMemo(
-    () => ({
-      showPoints: true,
-    }),
-    []
-  );
+  function extractLoadFromString(str = '') {
+    let list = str.split(',').map(ele => {
+      return ele.trim();
+    });
 
-  const axes = React.useMemo(
-    () => [
-      {
-        primary: true,
-        type: 'time',
-        position: 'bottom',
-      },
-      { type: 'linear', position: 'left' },
-    ],
-    []
-  );
+    let one_min = parseFloat(list[2].split(':')[1]);
+    let five_min = parseFloat(list[3]);
+    let fifteen_min = parseFloat(list[4]);
+
+    return {
+      x_value: new Date().getTime(),
+      y_value: [one_min, five_min, fifteen_min],
+      original_str: str,
+    };
+  }
+
+  function updateData(new_data) {
+    setstate(prevstate => ({
+      data: [
+        [...prevstate.data[0], new_data.y_value[0]],
+        [...prevstate.data[1], new_data.y_value[1]],
+        [...prevstate.data[2], new_data.y_value[2]],
+      ],
+      output: new_data.original_str,
+    }));
+  }
+
+  useEffect(() => {
+    const callback = () =>
+      exec(command)
+        .then(result => {
+          updateData(extractLoadFromString(result[0]));
+        })
+        .catch(err => console.log(err[1]));
+    const interval_handle = setInterval(callback, interval);
+    callback();
+    return () => clearInterval(interval_handle);
+  }, []);
 
   return (
     <>
       <Box
         style={{
-          padding: '0.5rem',
+          padding: '1.5rem',
         }}
       >
-        <br />
-        <Typography variant="overline" display="block" gutterBottom>
-          Camera Load averages
-        </Typography>
-        <br />
-
-        <ResizableBox
-          style={{
-            background: 'rgba(0, 27, 45, 0.9)',
-            borderRadius: '5px',
-            padding: '0.1rem',
-          }}
-        >
-          <Chart data={data} series={series} axes={axes} tooltip dark />
-        </ResizableBox>
+        <LoadChart data={state.data} title="Load averages" />
       </Box>
-      <Command command="uptime" interval={2000} updateData={updateData} />
+
+      <pre style={{ backgroundColor: '#eee', overflowX: 'auto' }}>
```

Comment:
Why to factor this out into something both `Command` and this widget can use?

Reply:
Hey, I have added a commit which shares data between graph and command.